### PR TITLE
Add summary styling to `KeyEventCard`

### PR DESCRIPTION
**Pull request by joecowton1**
## What does this change?

Adds an `isSummary` prop to the `KeyEventCard`, updates summary event styles and updates `decidePalette` to include new colours.

## Why?

Key event cards have different styling if an event is a summary.

Figma designs are here: https://www.figma.com/file/S0vY2PB6qxz797EeAE4vR6/liveblog-top-changes?node-id=566%3A22054

## Detail
- add an isSummary prop to the `KeyEventCard`
- use border-bottom for underline on @benwuersching 's instruction
- update underline on hover colour
- add new switches to `decidePalette` (wasn't 100% on where to add new fields in that file)
- update summary card bullet colour and bullet hover colour
- add new story to reflect summary card styles

### Before

https://user-images.githubusercontent.com/77005274/167663650-f7d19d3a-ba8f-4e94-a4ed-aa631ec1f6bb.mov

### After
https://user-images.githubusercontent.com/77005274/167663349-d491de26-7d19-4520-b4de-47f4823c142b.mov

### CODE REVIEW

File: dotcom-rendering/src/web/components/KeyEventCard.stories.tsx [hunk 1]
```diff
@@ -37,39 +37,69 @@ const wrapperStyles = css`
...
+		<StandardCard theme={ArticlePillar.Culture} count={1} />
+		<StandardCard theme={ArticlePillar.Lifestyle} count={1} />
+		<StandardCard theme={ArticlePillar.Sport} count={1} />
+		<StandardCard theme={ArticlePillar.Opinion} count={1} />
+		<StandardCard theme={ArticleSpecial.SpecialReport} count={1} />
 	</>
 );
 
-const MultipleCards = () => (
+const Multiple = () => (
```
Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventCard.stories.tsx:94]: Would it be worth adding a summary card into this?

File: dotcom-rendering/src/web/components/KeyEventCard.tsx [hunk 2]
```diff
@@ -9,39 +9,50 @@ export interface KeyEvent {
...
 	url: string;
 	format: ArticleFormat;
+	isSummary: boolean;
 }
 
-const linkStyles = css`
-	text-decoration: none;
-	line-height: 1.35;
+const linkStyles = (palette: Palette, isSummary: boolean) => {
+	const bulletColor = isSummary
```
Comment (abeddow91) [inline dotcom-rendering/src/web/components/KeyEventCard.tsx:16]: Nice and clear, thanks!

File: dotcom-rendering/src/web/components/KeyEventCard.tsx [hunk 3]
```diff
@@ -85,12 +96,22 @@ const timeStyles = css`
...
+	url,
+	format,
+	isSummary,
+}: KeyEvent) => {
 	const palette = decidePalette(format);
 
 	return (
 		<li css={listItemStyles}>
-			<Link priority="secondary" css={linkStyles} href={url}>
+			<Link
```
[the remark this answers is not in the document]
Comment (joecowton1) [reply dotcom-rendering/src/web/components/KeyEventCard.tsx:110]: Yeah i think this just needs main merging in - will do that now.

File: dotcom-rendering/src/web/components/KeyEventCard.tsx [hunk 4]
```diff
@@ -9,39 +9,50 @@ export interface KeyEvent {
...
+	isSummary: boolean;
 }
 
-const linkStyles = css`
-	text-decoration: none;
-	line-height: 1.35;
+const linkStyles = (palette: Palette, isSummary: boolean) => {
+	const bulletColor = isSummary
+		? palette.background.keyEventBullet
+		: neutral[46];
```
Comment (marjisound) [inline dotcom-rendering/src/web/components/KeyEventCard.tsx:18]: Would it be easy to bring the bullet colours for non-summary into the decide pallet too?

File: dotcom-rendering/src/web/components/KeyEventCard.stories.tsx (hunk 1 above)
Comment (joecowton1) [reply dotcom-rendering/src/web/components/KeyEventCard.stories.tsx:94]: The `events` data includes a `Summary` event so this is covered here - agree though its not obvious from the naming.

File: dotcom-rendering/src/web/components/KeyEventCard.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (abeddow91) [reply dotcom-rendering/src/web/components/KeyEventCard.tsx:110]: 👍

File: dotcom-rendering/src/web/components/KeyEventCard.tsx (hunk 4 above)
Comment (joecowton1) [reply dotcom-rendering/src/web/components/KeyEventCard.tsx:18]: Yeah maybe, at that point also the naming would need to better reflect what the colour is for here: ie `palette.background.keyEventSummaryBullet`. It's a bit long but maybe a better representation of what it actually is.

Comment (joecowton1) [reply dotcom-rendering/src/web/components/KeyEventCard.tsx]: Added all colours to `decidePalette` now 👍

Comment (marjisound) [reply dotcom-rendering/src/web/components/KeyEventCard.tsx]: Nice one 😄

**Review by marjisound — APPROVED** [review]
Looks great, well done 👍

**Pull request merged**